internal/driver/nodeserver_helpers.go

```diff
@@ -314,56 +316,55 @@ func (ns *NodeServer) mountVolume(ctx context.Context, devicePath string, req *c
 	// Format and mount the drive
 	log.V(4).Info("formatting and mounting the volume")
 	if err := ns.mounter.FormatAndMount(fmtAndMountSource, stagingTargetPath, fsType, mountOptions); err != nil {
-		return errInternal("Failed to format and mount device from (%q) to (%q) with fstype (%q) and options (%q): %v",
-			devicePath, stagingTargetPath, fsType, mountOptions, err)
+		return errInternal("Failed to format and mount device from (%q)---(%q) to (%q) with fstype (%q) and options (%q): %v",
+			fmtAndMountSource, devicePath, stagingTargetPath, fsType, mountOptions, err)
 	}
 
 	log.V(4).Info("Exiting mountVolume")
 	return nil
 }
 
-// prepareLUKSVolume prepares a LUKS-encrypted volume for mounting.
+// formatLUKSVolume prepares a LUKS-encrypted volume for mounting.
 //
 // It checks if the device at devicePath is already formatted with LUKS encryption.
 // If not, it formats the device using the provided LuksContext.
 // Finally, it prepares the LUKS volume for mounting.
-func (ns *NodeServer) prepareLUKSVolume(ctx context.Context, devicePath string, luksContext LuksContext) (string, error) {
+func (ns *NodeServer) formatLUKSVolume(ctx context.Context, devicePath string, luksContext LuksContext) (luksSource string, err error) {
 	log := logger.GetLogger(ctx)
-	log.V(4).Info("Entering prepareLUKSVolume", "devicePath", devicePath, "luksContext", luksContext)
+	log.V(4).Info("Entering formatLUKSVolume", "devicePath", devicePath, "luksContext", luksContext)
 
-	// LUKS encryption enabled, check if the volume needs to be formatted.
-	log.V(4).Info("LUKS encryption enabled")
+	// Validate the LUKS context.
+	if err = luksContext.validate(); err != nil {
+		return "", errInternal("Failed to luks format validation (%q): %v", devicePath, err)
+	}
 
-	// Validate if the device is formatted with LUKS encryption or if it needs formatting.
+	// LUKS encryption enabled, check if the volume needs to be formatted.
 	formatted, err := ns.encrypt.blkidValid(devicePath)
```

Comment:
nit: if possible could you add `blkidvalid()` to logging by passing the ctx in and updating it?

Reply:
done